internal/cmd/rsl/record/record.go

```diff
@@ -3,29 +3,58 @@
 package record
 
 import (
+	"fmt"
+
 	"github.com/gittuf/gittuf/internal/cmd/common"
+	"github.com/gittuf/gittuf/internal/dev"
 	"github.com/gittuf/gittuf/internal/repository"
 	"github.com/spf13/cobra"
 )
 
-type options struct{}
+type options struct {
+	fromEntry string
+}
 
-func (o *options) AddFlags(_ *cobra.Command) {}
+func (o *options) AddFlags(cmd *cobra.Command) {
+	cmd.Flags().StringVar(
+		&o.fromEntry,
+		"from-entry",
+		"",
+		fmt.Sprintf("perform verification from specified RSL entry (developer mode only, set %s=1)", dev.DevModeKey),
+	)
+}
 
-func (o *options) Run(_ *cobra.Command, args []string) error {
+func (o *options) Run(cmd *cobra.Command, args []string) error {
 	repo, err := repository.LoadRepository()
 	if err != nil {
 		return err
 	}
 
-	return repo.RecordRSLEntryForReference(args[0], true)
+	if err := repo.RecordRSLEntryForReference(args[0], true); err != nil {
+		return err
+	}
+
+	if o.fromEntry == "" {
+		err = repo.VerifyRef(cmd.Context(), args[0], false)
+	} else {
+		err = repo.VerifyRefFromEntry(cmd.Context(), args[0], o.fromEntry)
+	}
+
+	if err != nil {
+		if err := repo.InvalidateLatestRSLEntry(); err != nil {
+			return fmt.Errorf("removal of the RSL entry failed for %s: %w", args[0], err)
+		}
+		return fmt.Errorf("policy verification failed for %s: %w", args[0], err)
+	}
+	return nil
 }
 
 func New() *cobra.Command {
 	o := &options{}
 	cmd := &cobra.Command{
 		Use:               "record",
 		Short:             "Record latest state of a Git reference in the RSL",
+		Long:              "Record latest state of a Git reference in the RSL. Policy verification must succeed for the RSL entry to be created",
```

Comment:
```suggestion
		Long:              "Record latest state of a Git reference in the RSL. Policy verification must succeed for the RSL entry to be created.",
```

I'd actually make this a bit more verbose since this is the "long" message.

Reply:
Fixed. I don't think there is much more to say about it. Is there somthing else you would like me to add?